scripts/resource/Station.gd

```diff
@@ -42,15 +50,40 @@ func _init(p_id: int = 0, p_hydrogen: int = 0, p_power: int = 0, p_currency: int
 	crew = p_crew
 	rooms = p_rooms
 	time = p_time
+	update_max_hydrogen()
 
 func calculate_power_consumption_all_rooms() -> int:
-	var total = 0
+	var total_consuming_tiles = 0
+	var total_producing_tiles = 0
 	for room in rooms:
-		total += room.calculate_power_consumption()
-	return total
+		if room.roomType.id == 2: # 2 is the ID for generator rooms
+			total_producing_tiles += room.calculate_tile_count(room.topLeft, room.bottomRight)
+		else:
+			total_consuming_tiles += room.calculate_tile_count(room.topLeft, room.bottomRight)
+	return total_producing_tiles - total_consuming_tiles
 	
 func update_power() -> void:
-	power -= calculate_power_consumption_all_rooms()
+	power = calculate_power_consumption_all_rooms() * 100
 
 func update_hydrogen() -> void:
 	hydrogen += 5
+
+func update_max_hydrogen() -> void:
+	var total_storage_tiles = 0
+	for room in rooms:
+		if room.roomType.id == 3: # 3 is the ID for storage bays
+			total_storage_tiles += room.calculate_tile_count(room.topLeft, room.bottomRight)
+	max_hydrogen = 100 + (total_storage_tiles * 10)
+	
+func add_room(room: Room) -> void:
+	var new_rooms: Array[Room] = rooms.duplicate()
+	new_rooms.append(room)
+	rooms = new_rooms
```

Comment:
Sorry, I think I was not very clear when I was talking about this the other day. While this will work fine, I think we can just do all the logic here and then remove the set function above, which will remove the need for another function call and recreating the array each time.

```suggestion
	rooms.append(room)
	update_max_hydrogen()
	update_power()
```